pinot-spi/src/main/java/org/apache/pinot/spi/trace/DefaultRequestContext.java

```diff
@@ -63,6 +67,19 @@ public class DefaultRequestContext implements RequestScope {
 
   private FanoutType _fanoutType;
   private int _numUnavailableSegments;
+  private long _numConsumingSegmentsQueried;
```

Comment:
Not directly related to this PR -- I don't think we track RequestCompilationTime in BrokerResponse or RequestContext statistics. It is emitted as a metric from the code.

But I guess we should. Within LinkedIn we have always had requests that can take tens to hundreds of ms to compile a query. Will become more relevant in the context of multi stage engine imo.

Need not necessarily do as part of this PR though.

Reply:
Yes the plan is to add more metrics as need be. Thanks for bring this up - `RequestCompilationTime`. I am planning to add `stageStats` as well for multistage queries but that needs some discussion. Will take adding `RequestCompilationTime` in a follow-up PR. This PR tries to get alignment between brokerResponse and RequestContext class.